tools/python/boutiques/localExec.py

```diff
@@ -1047,6 +1047,68 @@ def _generateOutputFileNames(self):
                 outputFileName = os.path.abspath(outputFileName)
             self.out_dict[outputId] = outputFileName
 
+        # if 'conditional-path-template' in outputItem
+        # (key=conditions, value=path)
+        # Initialize file name with path template or existing value
+        for outputId in [x['id'] for x in self.outputs
+                         if 'conditional-path-template' in x]:
+
+            for boolObj in self.safeGet(outputId, 'conditional-path-template'):
+                # Surround expression chars with space
+                # Then split by space to isolate ids
+                boolExp = list(boolObj)[0]
```

Comment:
When turning a dictionary into a list, is it always ordered `['key', 'value']`?

Reply:
Had to test it :sweat_smile:, only the keys are taken when a dict is turned into a list. The reason for ```[0]``` is because there's always just one key:val per boolean object (type dict). 

I modified the code to be more readable: ```templateKey = list(templateObj.keys())[0]```